vllm/lora/ops/xla_ops/pallas.py

```diff
@@ -0,0 +1,131 @@
+# SPDX-License-Identifier: Apache-2.0
+import functools
+
+import jax
+import jax.numpy as jnp
```

Comment:
can you help me understand as to how the integration with torch xla and jax works? Last time I checked I couldn't share tensors between the two without a trip to the host. 
If the integration is solid we may re-use it in other parts.

Reply:
also this might be worth documenting anywhere even with a simple comment atop this file

Reply:
Ah I haven't heard about that. I based this off the `ragged_paged_attention` kernel work done recently. The pallas kernel/jax implementation is [here](https://github.com/pytorch/xla/blob/master/torch_xla/experimental/pallas_kernels/ragged_paged_attention_v2.py#L545) and the torch integration is [here](https://github.com/pytorch/xla/blob/master/torch_xla/experimental/custom_kernel.py#L994). 

I think the `make_kernel_from_pallas` glues jax/pt together avoiding a trip to the host.

Side note: I'm not sure if this is the best place to put these kernels, it could be the torch_xla repo too.